daisy/step_wait_for_instances_signal.go

```diff
@@ -178,6 +192,61 @@ func waitForSerialOutput(s *Step, project, zone, name string, so *SerialOutput,
 	}
 }
 
+func waitForGuestAttribute(s *Step, project, zone, name string, ga *GuestAttribute, interval time.Duration) DError {
+	w := s.w
+	msg := fmt.Sprintf("Instance %q: watching for key %s", name, ga.KeyName)
+	if ga.SuccessValue != "" {
+		msg += fmt.Sprintf(", SuccessValue: %q", ga.SuccessValue)
+	}
+	w.LogStepInfo(s.name, "WaitForInstancesSignal", msg+".")
+	var errs int
+	tick := time.Tick(interval)
+	for {
+		select {
+		case <-s.w.Cancel:
+			return nil
+		case <-tick:
+			resp, err := w.ComputeClient.GetGuestAttributes(project, zone, name, "", ga.KeyName)
+			if err != nil {
+				if apiErr, ok := err.(*googleapi.Error); ok && apiErr.Code == 404 {
+					// 404 is OK
+					continue
+				}
+				status, sErr := w.ComputeClient.InstanceStatus(project, zone, name)
+				if sErr != nil {
+					err = fmt.Errorf("%v, error getting InstanceStatus: %v", err, sErr)
+				} else {
+					err = fmt.Errorf("%v, InstanceStatus: %q", err, status)
+				}
+
+				// Wait until machine restarts to evaluate SerialOutput.
+				if status == "TERMINATED" || status == "STOPPED" || status == "STOPPING" {
+					continue
+				}
+
+				// Retry up to 3 times in a row on any error if we successfully got InstanceStatus.
+				if errs < 3 {
+					errs++
+					continue
+				}
+
```

Comment:
This, plus the other diff, ensures that you accumulate errors on failure to get the instance's status, and clear it when you *do* get the instance's status.

```suggestion
				// Retry up to 3 times in a row on any error if we successfully got InstanceStatus.
				if errs < 3 {
					continue
				}

```